database/src/main/postgres/runs/V1.5.13__create_or_update_additional_data.sql

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs.create_or_update_additional_data(
+    IN  i_partitioning      JSONB,
+    IN  i_additional_data   HSTORE,
+    IN  i_by_user           TEXT,
+    OUT status              INTEGER,
+    OUT status_text         TEXT,
+    OUT id_additional_data  BIGINT
+) RETURNS record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: runs.create_or_update_additional_data(3)
+--      Adds the additional data for the input partitioning. If additional data of a given name already
+--      exists for such partitioning, the value is updated and the old value is moved to the
+--      additional data history table.
+--
+-- Parameters:
+--      i_partitioning      - partitioning to add the additional data for
+--      i_additional_data   - sets of key/value pairs representing name and values of the additional data
+--      i_by_user           - user behind the change (an author of AD records if there will be something to upsert)
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_additional_data  - id of the data added
+--
+-- Status codes:
+--      11                  - Additional data have been added
+--      12                  - Additional data have been upserted
+--      14                  - No changes in additional data (this is when they already existed)
+--      41                  - Partitioning not found
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _fk_partitioning        BIGINT;
+    _ad_backup_performed    BOOLEAN;
+BEGIN
+
+    _fk_partitioning := runs._get_id_partitioning(i_partitioning, true);
```

Comment:
I don't actually like it that much:
a) if your IntelliJ is connected to the DB, then this is visible by the IntelliJ directly - no need to add it there
b) we (you, I, all of us) already did it like this on few places, including this exact function call
c) it could be a bit chaotic and non consistent - I would just not make a rule and leave it up to a personal style